modules/core-updates/9.0.md

```diff
@@ -117,6 +117,95 @@ Display event listeners for the back office in `dev` environment (default value
+    - `AdminController::displayAjax` (or specific `displayAjax{action}` method)
+    - `AdminController::display`
+      - init many variables used by the layout templates (meta title, toolbar, JS definitions, ...)
+      - render `page`, `header`, `footer` and sets them as Smarty variables
+      - fetch the Smarty `AdminController::template` to get the content of the page
+      - calls `smartyOutputContent` to echo/output the whole page (layout + content)
+
+**Symfony workflow:**
+
+- A special routing matching URLs like `?controller=LegacyAdminController` checks if the URL is valid thanks to `LegacyRouterChecker::check`
+  - the query parameter must match an existing legacy controller class
+  - the controller object is instanciated and `AdminController::init` is called, this is important to know if the controller is anonymous, this is the proper moment to define `AdminController::allowAnonymous`
+  - a few attributes are set on the Symfony `Request` so they can be reused later in the workflow by other components/services
+- `LegacyController::legacyPageAction` is a generic controller responsible for executing the legacy controller, it doesn't execute `AdminController::run` but splits it into several steps
+  - `AdminController::checkAccess` is not called, the token in URLs is a Symfony CSRF token now and is checked by `TokenizedUrlListener`
+  - `LegacyController::initController`
+    - the permission is checked based on the detected controller and action, unless `AdminController::viewAccess` has been overridden in which case it is used to check authorization
+    - `SmartyVariablesFiller::fillDefault` is in charge of building most of the generic smarty variables previously handled by `AdminController::initHeader`
+    - `AdminController::setMedia` is in charge of defining the JS/CSS files to be loaded in the page
+    - `AdminController::postProcess` processes the posted data
+      - if after this method `AdminController::redirect_after` is defined then perform an HTTP redirection
+  - Check if `ajax` query parameter is defined, one of the two is executed
+    - `LegacyController::renderAjaxController` executes `AdminController::initContent` the appropriate `AdminController::displayAjax` method, it catches all the output data into a string buffer (because most legacy ajax methods simply echo their content) and returns it as a response
+    - `LegacyController::renderPageContent`
+      - executes `AdminController::initContent`
+      - fetch the Smarty `AdminController::template` to get the content of the page
+      - calls `Cookie::write` which was previously called via `smartyOutputContent`
+      - get the modal content via `AdminController::renderModal`
+
+The two workflows should render the same result, many methods from the legacy workflow are not executed anymore because they lost their purpose, but in case you overrode one of those methods be aware that they are no longer called directly so they are breaking changes:
```

Comment:
```suggestion
The two workflows should render the same result. Many methods from the legacy workflow are not executed anymore because they lost their purpose, but in case you override one of those methods, be aware that they are no longer called directly, so there are breaking changes:
```